agent/xds/routes.go

```diff
@@ -874,6 +874,10 @@ func (s *ResourceGenerator) makeRouteActionForSplitter(
 		clusters = append(clusters, cw)
 	}
 
+	if len(clusters) <= 0 {
+		return nil, fmt.Errorf("number of clusters in splitter must be > 0; got %d", len(clusters))
```

Comment:
Did you experience an issue that warranted this?

Reply:
Initially, when I saw some error message from envoy, saying cluster must not be empty for route configuration, I thought this might be the root cause. However, it turns out this will always occur and proxycfg will reconcile and add the cluster to the router configuration.

But I leave it here to avoid sending a configuration with empty cluster to envoy since envoy will always return an error message.